File: 2/solution.rb
fib_list = [1, 2]

def populate_fib_list(fib_list)
  next_num = fib_list[-1] + fib_list[-2]

  unless next_num > 4000000
    fib_list.push(next_num)
    populate_fib_list(fib_list)
  end
  even_num_sum_finder(fib_list)
end

def even_num_sum_finder(fib_list)
  sum = 0

  fib_list.each do |i|
    if i % 2 == 0
      sum += i
    end
  end
  sum
end

File: 2/solution.js
var fibList = [1, 2];

function populateFibList() {
  var nextNum = fibList[fibList.length - 1] + fibList[fibList.length - 2];

  if (nextNum > 4000000) {
    evenNumSumFinder(fibList);
  } else {
    fibList.push(nextNum);
    populateFibList();
  }
}

function evenNumSumFinder(fibList) {
  var sum = 0;
  for (var i = 0; i < fibList.length; i++) {
    if (fibList[i] % 2 === 0) {
      sum += fibList[i];
    }
  }
  return sum;
}
